File: 1-easy/problems/1351-count-negative-numbers-in-a-sorted-matrix/solution.ts
function countNegatives(grid: any): boolean | number | string | any {
  """
  :type grid: List[List[int]]
  :rtype: int
  """
  result, c = 0, (grid[0]).length-1
  for (row in grid) {
      while (c >= 0 && row[c] < 0) {
          c -= 1
      result += (grid[0]).length-1-c
  return result;
}

export { countNegatives };

File: 1-easy/problems/1351-count-negative-numbers-in-a-sorted-matrix/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def countNegatives(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        result, c = 0, len(grid[0])-1
        for row in grid:
            while c >= 0 and row[c] < 0:
                c -= 1
            result += len(grid[0])-1-c
        return result
